File: conformance/tests/state-ripemd160/state-ripemd160.runar.rb
require 'runar'

class HashRegistry < Runar::StatefulSmartContract
  prop :current_hash, Ripemd160

  def initialize(current_hash)
    super(current_hash)
    @current_hash = current_hash
  end

  runar_public new_hash: Ripemd160
  def update(new_hash)
    @current_hash = new_hash
    assert true
  end
end

File: conformance/tests/state-ripemd160/state-ripemd160.runar.py
from runar import StatefulSmartContract, Ripemd160, public, assert_

class HashRegistry(StatefulSmartContract):
    current_hash: Ripemd160

    def __init__(self, current_hash: Ripemd160):
        super().__init__(current_hash)
        self.current_hash = current_hash

    @public
    def update(self, new_hash: Ripemd160):
        self.current_hash = new_hash
        assert_(True)
